### ISSUE TITLE
Bundle Update fail around `http_parser.rb 0.6.0`

### ISSUE THREAD

**Issue by kekoziar** [rubygems/rubygems#4771]
### Describe the problem as clearly as you can

I have an ongoing `bundle update` fail around `http_parser.rb 0.6.0`.  I suspect it is because there is a space in my user directory folder name.  The computer I am trying to fix is administered by my department, and so I can't control that there is a space in my user directory folder name.  When I run `bundle update` on a personal computer that I have control over, and so have no spaces in my path, it runs fine.  I'm not sure if this problem has been fixed in `http_parser.rb 0.7.0`, but if it has, perhaps `bundle update` can refer to the updated `http_parser.rb`.

```
Fetching http_parser.rb 0.6.0
Installing http_parser.rb 0.6.0 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
C:/Ruby30-x64/bin/ruby.exe -I C:/Ruby30-x64/lib/ruby/3.0.0 -r
./siteconf20210720-2696-dc5zry.rb extconf.rb
creating Makefile

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make DESTDIR\= clean

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make DESTDIR\=
generating ruby_http_parser-x64-mingw32.def
compiling ruby_http_parser.c
gcc.exe: error:
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser:
No such file or directory
make: *** [Makefile:245: ruby_http_parser.o] Error 1
```

### Post steps to reproduce the problem

1. Put a space in your path.  
2. Run `bundle update`

Ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x64-mingw32]
Installed from https://rubyinstaller.org/

Bundler version 2.2.24
Installed via git-bash CLI (`gem install bundler`)

Git-bash version 2.32.0

### Which command did you run?

`bundle update`

### What were you expecting to happen?

Bundler to update. 

### What actually happened?
```
$ bundle update
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Fetching concurrent-ruby 1.1.9
Installing concurrent-ruby 1.1.9
Fetching i18n 0.9.5
Installing i18n 0.9.5
Fetching minitest 5.14.4
Installing minitest 5.14.4
Fetching thread_safe 0.3.6
Installing thread_safe 0.3.6
Fetching tzinfo 1.2.9
Installing tzinfo 1.2.9
Fetching zeitwerk 2.4.2
Installing zeitwerk 2.4.2
Fetching activesupport 6.0.4
Installing activesupport 6.0.4
Fetching public_suffix 4.0.6
Installing public_suffix 4.0.6
Fetching addressable 2.8.0
Installing addressable 2.8.0
Using bundler 2.2.24
Fetching coffee-script-source 1.11.1
Installing coffee-script-source 1.11.1
Fetching execjs 2.8.1
Installing execjs 2.8.1
Fetching coffee-script 2.4.1
Installing coffee-script 2.4.1
Fetching colorator 1.1.0
Installing colorator 1.1.0
Fetching ruby-enum 0.9.0
Installing ruby-enum 0.9.0
Fetching commonmarker 0.17.13
Installing commonmarker 0.17.13 with native extensions
Fetching unf_ext 0.0.7.7
Installing unf_ext 0.0.7.7 with native extensions
Fetching unf 0.1.4
Installing unf 0.1.4
Fetching simpleidn 0.2.1
Installing simpleidn 0.2.1
Fetching dnsruby 1.61.7
Installing dnsruby 1.61.7
Fetching eventmachine 1.2.7 (x64-mingw32)
Installing eventmachine 1.2.7 (x64-mingw32)
Fetching http_parser.rb 0.6.0
Installing http_parser.rb 0.6.0 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
C:/Ruby30-x64/bin/ruby.exe -I C:/Ruby30-x64/lib/ruby/3.0.0 -r
./siteconf20210720-2696-dc5zry.rb extconf.rb
creating Makefile

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make DESTDIR\= clean

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make DESTDIR\=
generating ruby_http_parser-x64-mingw32.def
compiling ruby_http_parser.c
gcc.exe: error:
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser:
No such file or directory
make: *** [Makefile:245: ruby_http_parser.o] Error 1

make failed, exit code 2

Gem files will remain installed in C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0 for
inspection.
Results logged to C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/extensions/x64-mingw32/3.0.0/http_parser.rb-0.6.0/gem_make.out

An error occurred while installing http_parser.rb (0.6.0), and Bundler cannot
continue.
Make sure that `gem install http_parser.rb -v '0.6.0' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  github-pages was resolved to 215, which depends on
    jekyll-avatar was resolved to 0.7.0, which depends on
      jekyll was resolved to 3.9.0, which depends on
        em-websocket was resolved to 0.5.2, which depends on
          http_parser.rb
```
### If not included with the output of your command, run `bundle env` and paste the output below
````
$ bundle env
## Environment

```
Bundler       2.2.24
  Platforms   ruby, x64-mingw32
Ruby          3.0.2p107 (2021-07-07 revision 0db68f023372b634603c74fca94588b457be084c) [x64-mingw32]
  Full Path   C:/Ruby30-x64/bin/ruby.exe
  Config Dir  C:/ProgramData
RubyGems      3.2.22
  Gem Home    C:/Ruby30-x64/lib/ruby/gems/3.0.0
  Gem Path    C:/Users/firstname lastname/.gem/ruby/3.0.0;C:/Ruby30-x64/lib/ruby/gems/3.0.0
  User Home   C:/Users/firstname lastname
  User Path   C:/Users/firstname lastname/.gem/ruby/3.0.0
  Bin Dir     C:/Ruby30-x64/bin
Tools
  Git         2.32.0.windows.1
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2021-07-15
Git SHA           d78b1ee235
Released Version  true
```

**Comment by indirect** [comment]
This is a common problem with gems that compile native extensions, and usually has nothing to do with Bundler. You can check by running `gem install http_parser --version 0.6.0`. If that fails, the problem is with the gem and not with Bundler. If that succeeds, please come back and let us know so we can investigate.

**Comment by kekoziar** [comment]
@indirect  `gem install http_parser.rb --version 0.6.0` runs fine (see below).  But, when I run `bundle update` so I can finish updating all of the gems, it still fails at `http_parser.rb (0.6.0)`

```
$ gem install http_parser.rb --version 0.6.0
unknown encoding name "chunked\r\n\r\n25" for ext/ruby_http_parser/vendor/http-parser-java/tools/parse_tests.rb, skipping
Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions. This could take a while...
Successfully installed http_parser.rb-0.6.0
Parsing documentation for http_parser.rb-0.6.0
Installing ri documentation for http_parser.rb-0.6.0
Done installing documentation for http_parser.rb after 1 seconds
1 gem installed
```

**Comment by indirect** [comment]
Please see https://github.com/tmm1/http_parser.rb/issues/47, where it is revealed that this _is_ an issue with http_parser, and that it is fixed in 0.6.1.

**Comment by kekoziar** [comment]
@indirect when will `bundle update` use 0.6.1 instead of 0.6.0?  This solution you posted is 4 years old.  The current http_parser.rb version is 0.7.0.

**Comment by indirect** [comment]
It will happen as soon as you allow it. I have no idea what is in your Gemfile ensuring you get 0.6.0, you'll have to figure that out yourself. You could try changing your Gemfile to include (at least) `gem "http_parser", ">= 0.6.1"` as a first step.

**Comment by kekoziar** [comment]
Thank you very much for your feedback and assistance on this, @indirect.  I don't have a lot in my gemfile, just two lines: `source 'https://rubygems.org';  gem 'github-pages', group: :jekyll_plugin`.  I tracked down the culprit.  It's a gem listed on rubygems.org (where bundler pulls the metadata) github-pages -> jekyll 3.9.0 = 3.9.0 -> em-websocket 0.5.2 ~> 0.5 -> http_parser.rb 0.6.0 ~> 0.6.0, then opened an [issue](https://github.com/igrigorik/em-websocket/issues/157) with the em-websocket repo.  

Adding the specific gem and version to my Gemfile does solve the problem, and `bundle update` ran with no errors.

**Comment by harriott** [comment]
Deleting `Gemfile.lock` fixed it for me - no idea why...

**Comment by Ankit9721** [comment]
Deleting Gemfile.lock fixed it for me - no idea why...

**Comment by Ankit9721** [comment]
$ bundle update
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Fetching concurrent-ruby 1.1.9
Installing concurrent-ruby 1.1.9
Fetching i18n 0.9.5
Installing i18n 0.9.5
Fetching minitest 5.14.4
Installing minitest 5.14.4
Fetching thread_safe 0.3.6
Installing thread_safe 0.3.6
Fetching tzinfo 1.2.9
Installing tzinfo 1.2.9
Fetching zeitwerk 2.4.2
Installing zeitwerk 2.4.2
Fetching activesupport 6.0.4
Installing activesupport 6.0.4
Fetching public_suffix 4.0.6
Installing public_suffix 4.0.6
Fetching addressable 2.8.0
Installing addressable 2.8.0
Using bundler 2.2.24
Fetching coffee-script-source 1.11.1
Installing coffee-script-source 1.11.1
Fetching execjs 2.8.1
Installing execjs 2.8.1
Fetching coffee-script 2.4.1
Installing coffee-script 2.4.1
Fetching colorator 1.1.0
Installing colorator 1.1.0
Fetching ruby-enum 0.9.0
Installing ruby-enum 0.9.0
Fetching commonmarker 0.17.13
Installing commonmarker 0.17.13 with native extensions
Fetching unf_ext 0.0.7.7
Installing unf_ext 0.0.7.7 with native extensions
Fetching unf 0.1.4
Installing unf 0.1.4
Fetching simpleidn 0.2.1
Installing simpleidn 0.2.1
Fetching dnsruby 1.61.7
Installing dnsruby 1.61.7
Fetching eventmachine 1.2.7 (x64-mingw32)
Installing eventmachine 1.2.7 (x64-mingw32)
Fetching http_parser.rb 0.6.0
Installing http_parser.rb 0.6.0 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
C:/Ruby30-x64/bin/ruby.exe -I C:/Ruby30-x64/lib/ruby/3.0.0 -r
./siteconf20210720-2696-dc5zry.rb extconf.rb
creating Makefile

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make DESTDIR\= clean

current directory: C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make DESTDIR\=
generating ruby_http_parser-x64-mingw32.def
compiling ruby_http_parser.c
gcc.exe: error:
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser:
No such file or directory
make: *** [Makefile:245: ruby_http_parser.o] Error 1

make failed, exit code 2

Gem files will remain installed in C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/gems/http_parser.rb-0.6.0 for
inspection.
Results logged to C:/Users/firstname
lastname/repos/git-novice/.vendor/bundle/ruby/3.0.0/extensions/x64-mingw32/3.0.0/http_parser.rb-0.6.0/gem_make.out

An error occurred while installing http_parser.rb (0.6.0), and Bundler cannot
continue.
Make sure that `gem install http_parser.rb -v '0.6.0' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  github-pages was resolved to 215, which depends on
    jekyll-avatar was resolved to 0.7.0, which depends on
      jekyll was resolved to 3.9.0, which depends on
        em-websocket was resolved to 0.5.2, which depends on
          http_parser.rb
